packages/@react-stately/table/src/TableUtils.ts

```diff
@@ -57,118 +57,267 @@ export function getMinWidth(minWidth: number | string, tableWidth: number): numb
     : 0;
 }
 
-// tell us the delta between min width and target width vs max width and target width
-function mapDynamicColumns(dynamicColumns: IndexedColumn[], availableSpace: number, tableWidth: number): (IndexedColumn & {delta: number})[] {
-  let fractions = dynamicColumns.reduce(
-    (sum, column) => column ? sum + parseFractionalUnit((column.column.width || column.column.defaultWidth) as string) : sum,
-    0
-  );
-
-  let columns = dynamicColumns.map((column) => {
-    if (!column) {
-      return null;
+
+export interface IColumn {
+  minWidth?: number | string,
+  maxWidth?: number | string,
+  width?: number | string,
+  defaultWidth?: number | string,
+  key?: Key
+}
+
+/**
+ * Implements the flex algorithm described in https://www.w3.org/TR/css-flexbox-1/#layout-algorithm
+ * It makes a few constraint/assumptions:
+ * 1. All basis values are 0 unless it is a static width, then the basis is the static width
+ * 2. All flex grow and shrink values are equal to the FR specified on the column, grow and shrink for the same column are equal
+ * 3. We only have one row
+ * An example of the setup can be seen here https://jsfiddle.net/snowystinger/wv0ymjaf/61/ where I let the browser figure out the
+ * flex of the columns.
+ * Note: We differ in one key aspect, all of our column widths must be whole numbers, so we avoid browser
+ * sub pixel rounding errors. To do this, we use a cascading rounding algorithm to ensure that the sum of the widths is maintained
+ * while distributing the rounding remainder across the columns.
+ *
+ * As noted in the chrome source code, this algorithm is very accurate, but has the potential to be quadratic.
+ * They have deemed this to be acceptable because the number of elements is usually small and the flex factors
```

Comment:
> number of elements is usually small

is this true in our case with a large number of columns?

Reply:
I believe it to be true, I don't think we'll often find a table with 1fr, 2fr, 3fr, 4fr, 5fr and so on all different, it's more common that it'll all be 1fr. Same thing with minWidths, it's unlikely that there will be large variety of them.
In addition, once the sum of column widths exceeds the visible table viewport width, things simplify because at that point everything is at its min width or it's static. So nothing needs to be done.